### PR TITLE
Add model DSTAGNN

### PR DESCRIPTION
**Pull request by hczs**
## 任务执行配置信息
数据集文件配置（DSTAGNNDataset.json），跑 PEMSD4 数据集时要调整 batch_size 参数为 32
```json
{
  "batch_size": 64,
  "cache_dataset": true,
  "num_workers": 0,
  "train_rate": 0.6,
  "eval_rate": 0.2,
  "scaler": "standard",
  "load_external": false,
  "normal_external": false,
  "ext_scaler": "none",
  "input_window": 12,
  "output_window": 12,
  "sparsity": 0.01,
  "period": 288
}
```
模型参数配置（DSTAGNN.json）
```json
{
  "max_epoch": 100,

  "learner": "adam",
  "learning_rate": 0.0001,
  "use_early_stop": false,

  "in_channels": 1,
  "nb_block": 4,
  "K": 3,
  "nb_chev_filter": 32,
  "nb_time_filter": 32,
  "d_model": 512,
  "d_k": 32,
  "n_heads": 3,
  "graph_use": "AG"
}
```
## 任务执行结果
模型评价指标对比如下，DSTAGNN 的指标是使用原论文代码本地复现的结果
- PEMSD4
  | PEMSD4 | LibCity | DSTAGNN |
  | ---- | ---- | ---- |
  |  MAE |  19.63 | 19,28 |
  | RMSE |  31.89 |  31.37 |
  | MAPE |  13.15% |  12.72% |

- PEMSD8
  | PEMSD8 | LibCity | DSTAGNN |
  | ---- | ---- | ---- |
  |  MAE |  16.11 | 15.81 |
  | RMSE |  25.15 |  25.03 |
  | MAPE |  10.44% |  9.92% |

### CODE REVIEW

File: libcity/model/loss.py [hunk 1]
```diff
@@ -157,3 +157,8 @@ def explained_variance_score_np(preds, labels):
     preds = preds.flatten()
     labels = labels.flatten()
     return explained_variance_score(labels, preds)
+
+
+def smooth_l1_loss(preds, labels):
+    loss_func = torch.nn.SmoothL1Loss()
```
Comment (Kazeya27) [inline libcity/model/loss.py:163]: smooth_l1_loss需要按官方公式重新实现，类似huber_loss

**Review by Kazeya27 — CHANGES_REQUESTED** [review]

File: libcity/model/loss.py (hunk 1 above)
Comment (hczs) [reply libcity/model/loss.py]: 已经按照官方公式重新实现：https://github.com/LibCity/Bigscity-LibCity/pull/406/commits/b66bbff3f01893b4fab4810cd5ea0f0d306f0b05

**Review by Kazeya27 — APPROVED** [review]
